src/main/java/org/commonjava/cdi/util/weft/ThreadContext.java

```diff
@@ -106,10 +106,10 @@ private static ThreadContext swapContext( final ThreadContext ctx )
      */
     public synchronized void registerFinalizer( Consumer<ThreadContext> finalizer )
     {
-        Logger logger = LoggerFactory.getLogger( getClass() );
-        logger.trace( "Registering finalizer: {} on ThreadContext: {}", finalizer, this );
         if ( !this.finalizers.contains( finalizer ) )
         {
+            Logger logger = LoggerFactory.getLogger( getClass() );
+            logger.debug( "Registering finalizer: {} on ThreadContext: {}", finalizer, this );
```

Comment:
We might want to put this back on TRACE log level, but not a huge problem for now.